### ISSUE TITLE
Style/RedundantLineContinuation: false positive

### ISSUE THREAD

**Issue by akimd** [rubocop/rubocop#13145]
Hi,

In the following example the backslash is absolutely needed, yet the cop wants to get rid of it.

```
$ cat backslash.rb
puts foo(123,
         123) \
     == bar
```
```
$ bundle exec rubocop backslash.rb
Inspecting 1 file
C

Offenses:

backslash.rb:1:1: C: [Correctable] Style/FrozenStringLiteralComment: Missing frozen string literal comment.
puts foo(123,
^
backslash.rb:2:15: C: [Correctable] Style/RedundantLineContinuation: Redundant line continuation.
         123) \ ...
              ^

1 file inspected, 2 offenses detected, 2 offenses autocorrectable
```

Cheers!

--------

## Expected behavior

No complaint.

## Actual behavior

Complaint.

## RuboCop version

```
$ bundle exec rubocop -V
1.65.1 (using Parser 3.3.4.2, rubocop-ast 1.32.1, running on ruby 3.3.4) [arm64-darwin23]
  - rubocop-performance 1.21.1
```

**Comment by koic** [comment]
This is a false positive, not a false negative.

**Comment by akimd** [comment]
Of course, sorry.

**Comment by akimd** [comment]
Another example of false positive.  I don't know if it's the same bug.
```ruby
# frozen_string_literal: true

def fun
  foo
    .bar(12) \
  || baz
end
```

```
$ bundle exec rubocop contline.rb
Inspecting 1 file
C

Offenses:

contline.rb:5:14: C: [Correctable] Style/RedundantLineContinuation: Redundant line continuation.
    .bar(12) \ ...
             ^

1 file inspected, 1 offense detected, 1 offense autocorrectable
```
```
$ bundle exec rubocop -V
1.66.1 (using Parser 3.3.5.0, rubocop-ast 1.32.3, running on ruby 3.3.4) [arm64-darwin23]
  - rubocop-performance 1.21.1
```

**Comment by adityasinghbaghel** [comment]
Hey, @koic  Can I work on this issue?

**Comment by adityasinghbaghel** [comment]
@akimd 

> Another example of false positive. I don't know if it's the same bug.
> 
> ```ruby
> # frozen_string_literal: true
> 
> def fun
>   foo
>     .bar(12) \
>   || baz
> end
> ```
> 
> ```
> $ bundle exec rubocop contline.rb
> Inspecting 1 file
> C
> 
> Offenses:
> 
> contline.rb:5:14: C: [Correctable] Style/RedundantLineContinuation: Redundant line continuation.
>     .bar(12) \ ...
>              ^
> 
> 1 file inspected, 1 offense detected, 1 offense autocorrectable
> ```
> 
> ```
> $ bundle exec rubocop -V
> 1.66.1 (using Parser 3.3.5.0, rubocop-ast 1.32.3, running on ruby 3.3.4) [arm64-darwin23]
>   - rubocop-performance 1.21.1
> ```

In this case it is not a issue as Ruby allows to chain method calls across multiple lines without needing a backslash if the line continues naturally.

**Comment by akimd** [comment]
@adityasinghbaghel I think you misread my example.  The continuation is not on a method call, but on the `||`.

**Comment by koic** [comment]
FYI, I've opened #13283 to solve this issue.

**Comment by akimd** [comment]
https://github.com/rubocop/rubocop/issues/13145#issuecomment-2338703732 was not addressed.  I'm going to open another issue.
